packages/backend/src/plugins/events.ts

```diff
@@ -35,8 +33,8 @@ export default async function createPlugin(
   http.bind(eventsRouter);
 
   await new EventsBackend(env.logger)
+    .setEventBroker(env.eventBroker)
     .addPublishers(http)
```

Comment:
@pjungermann if we go down the route of making what is currently the `EventBroker` interface more or less directly available as the core `EventService`, do you think it makes sense to scrap the `EventPublisher` interface? I'm thinking that the `setEventBroker` indirection won't be necessary if you have access to the `publish` method directly, but is that breaking some abstraction that you think it's important to have in place?

Reply:
Id also be curious about this.

Reply:
Yes, likely we could drop the `EventPublisher` interface.

You already have access to the `publish` method directly, though. Event publishers only receive the event broker via `setEventBroker` (after initialization; similar to the connection e.g. at `EntityProvider`) and then call the `publish` method provided by it.

However, as we move the `EventBroker` to the env now, we can pass it directly to the constructor of an EventPublisher -- or whatever is suitable for the implementation. No need to force the `setEventBroker` way.

This would work already after this change. It is a breaking change for the current EventPublisher users, though.

The open question is, how we want to change it further to have the `EventService`. Probably better to do it as part of the following changes.

Reply:
Might make sense to mark the `EventPublisher` interface as deprecated already.

Reply:
I think there is enough in this commit for the moment. Do you mind if we mark the deprecation in a subsequent commit?

Reply:
I dont quite understand when the sqs implemention is an EventPublisher and not an EventBroker. It seems to me like it should publish and subscribe to the same infrastructure? Am I missing a piece here?

Reply:
> I dont quite understand when the sqs implemention is an EventPublisher and not an EventBroker.

In general, it could receive events, too, sure. Often, you have SNS or EventBridge before the SQS queue.

In the context here, it is only an alternative to the HTTP way of ingesting events into Backstage. From the Backstage perspective, it is a mere publisher of events.

It is not subscribed to anything, no subscriber. And you cannot publish events to it or subscribe to specific topics at it, so no broker either.

We use it to deliver webhook events to Backstage without having Backstage accessible/used as subscription destination at the webhook event producer, and events are buffered and pulled vs pushed as at the HTTP-based version.

You could write your own EventBroker implementation, though, using any of these AWS services or Kafka or whatever makes sense.